xtra-libp2p/src/dialer.rs

```diff
@@ -2,38 +2,37 @@ use crate::endpoint;
 use crate::multiaddress_ext::MultiaddrExt;
 use crate::Connect;
 use crate::Endpoint;
+use crate::GetConnectionStats;
 use anyhow::anyhow;
+use anyhow::Result;
 use async_trait::async_trait;
 use libp2p_core::Multiaddr;
 use libp2p_core::PeerId;
 use std::time::Duration;
-use tokio_tasks::Tasks;
 use xtra::Address;
 use xtra_productivity::xtra_productivity;
+use xtras::SendAsyncSafe;
 
 /// If we're not connected by this time, stop the actor.
-pub const CONNECTION_TIMEOUT: Duration = Duration::from_secs(5);
+pub const CONNECTION_TIMEOUT: Duration = Duration::from_secs(20);
```

Comment:
I don't think this is necessary. We should potentially try more often then every `20` seconds. 

If we trigger this and the endpoint is still waiting the endpoint will just return a 
But it's fine for the dialer to check again after 5 secs.

Aligning this could mean that retrying to connect takes 20 seconds.

Consider this:

1. The dialer triggers a `Connect` in the endpoint.
2. The endpoint fails to connect after 6 seconds. 
3. The dialer restarts only after 20... 

If the dialer would restart with the 5 secs it would catch such a scenario faster, and it should not do harm.

Reply:
I will revert the last commit, as 5s was catching the scenario faster; as such, it probably should have not been included in this PR.